Glucose conversion command

p5nbTgip0r opened the pull request:
<details>
  <summary>Preview</summary>
  <img width="586" height="1239" alt="image" src="https://github.com/user-attachments/assets/0197929a-4f14-4da6-985f-efe8f1dd8e4c" />
</details>
Adds a /convert command, closes #2.

p5nbTgip0r committed:
feat: add glucose unit conversion command with tests

p5nbTgip0r committed:
test: add display tests for Glucose

cascer1 reviewed (changes requested):
Thanks for your PR! It mostly looks good but I've got some architectural questions. I'm not very used to reading Rust (yet) but I think you went with some shorthand approaches in several places where perhaps separate function calls and checks would improve readability.

If you disagree, let me know.

p5nbTgip0r committed:
refactor: add support for separate unit parameter, support commas, simplify parsing code

p5nbTgip0r committed:
refactor: extract common color constants, add success color constant

p5nbTgip0r committed:
refactor: move glucose tests into same module

p5nbTgip0r committed:
refactor: relax precision threshold in glucose comparison function

For glucose values, we don't need a precision of 0.00001; 0.001 is plenty.

p5nbTgip0r committed:
refactor: hardcode expected values in glucose conversion tests instead of calculating them

p5nbTgip0r committed:
feat: add optional unit parameter to convert command

p5nbTgip0r committed:
refactor: use f32 instead of f64 for mmol glucose values

p5nbTgip0r committed:
fix: allow negative glucose values

p5nbTgip0r committed:
feat: clamp glucose values between -9999 and 9999

p5nbTgip0r committed:
refactor: replace `as_` prefixes to `to_` to match rust api guidelines

https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv

p5nbTgip0r committed:
refactor: suppress dead code warning for `SUCCESS` color

p5nbTgip0r committed:
refactor: fix clippy lint warnings

p5nbTgip0r committed:
docs: specify `to_mgdl` and `to_mmol` return `self` instead of a clone
